inc/class_core.php

```diff
@@ -277,14 +277,18 @@ function __construct()
 		$this->parse_incoming($_GET);
 		$this->parse_incoming($_POST);
 
-		if($_SERVER['REQUEST_METHOD'] == "POST")
+		if(isset($_SERVER['REQUEST_METHOD']) && $_SERVER['REQUEST_METHOD'] == "POST")
 		{
 			$this->request_method = "post";
 		}
-		else if($_SERVER['REQUEST_METHOD'] == "GET")
+		else if(isset($_SERVER['REQUEST_METHOD']) && $_SERVER['REQUEST_METHOD'] == "GET")
 		{
 			$this->request_method = "get";
 		}
+		else
+		{
+			$this->request_method = "";
```

Comment:
Wouldn't it make sense to default to "get"? MyBB doesn't support other request methods anyway.

Reply:
Probably, I just went with the logic around it.